include/aws/common/json.h

```diff
@@ -0,0 +1,395 @@
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_node_new(void);
+
+/**
+ * Creates a new string aws_json_node with the given string and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * @param string The string you want to store in the aws_json_node
+ * @return A new string aws_json_node
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_string_new(char *string);
+
+/**
+ * Creates a new number aws_json_node with the given number and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * @param number The number you want to store in the aws_json_node
+ * @return A new number aws_json_node
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_number_new(double number);
+
+/**
+ * Creates a new array aws_json_node and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * Deleting this array will also delete any aws_json_nodes it contains.
+ * @return A new array aws_json_node
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_array_new(void);
+
+/**
+ * Creates a new boolean aws_json_node with the given boolean and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * @param boolean The boolean you want to store in the aws_json_node
+ * @return A new boolean aws_json_node
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_boolean_new(bool boolean);
+
+/**
+ * Creates a new null aws_json_node and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * @return
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_null_new(void);
+
+/**
+ * Creates a new object aws_json_node and returns a pointer to it.
+ *
+ * Note: You will need to free the memory for the aws_json_node using aws_json_delete on the aws_json_node or
+ * on the object/array containing the aws_json_node.
+ * Deleting this object will also delete any aws_json_nodes it contains.
+ * @return
+ */
+AWS_COMMON_API
+struct aws_json_node *aws_json_object_new(void);
+// ====================
+
+// ====================
+// Value setters and getters
+
+/**
+ * Sets the string value of a string aws_json_node.
+ * @param node The string aws_json_node.
+ * @param string The new string value.
+ * @return AWS_OP_SUCCESS if the string was successfully set, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_string_set(struct aws_json_node *node, char *string);
```

Comment:
we might not even want to offer setters for basic values, just set the value initially in the new() function

this just needs to be the most basic possible API